yaml_test.go

```diff
@@ -411,8 +411,8 @@ func TestUnmarshal(t *testing.T) {
 		// decoding integers
 		"decode 2^53 + 1 into int": {
 			encoded:    []byte("9007199254740993"),
-			decodeInto: new(int),
-			decoded:    9007199254740993,
+			decodeInto: new(int64),
+			decoded:    int64(9007199254740993),
```

Comment:
if we want to *add* a test that explicitly decodes into an int64, that's fine

let's try to keep this testing a decode into an unsized int on 64-bit platforms

can do that with helpers that use strconv.IntSize to return a new `*int` or `*int64`, and cast the expected result to int64 on 32-bit platforms